demo/src/main/java/com/google/android/exoplayer2/demo/player/RtmpDataSource.java

```diff
@@ -0,0 +1,55 @@
+package com.google.android.exoplayer2.demo.player;
+
+
+import android.net.Uri;
+
+import com.google.android.exoplayer2.C;
+import com.google.android.exoplayer2.upstream.DataSource;
+import com.google.android.exoplayer2.upstream.DataSpec;
+
+import net.butterflytv.rtmp_client.RtmpClient;
+
+import java.io.IOException;
+
+/**
+ * Created by faraklit on 08.01.2016.
+ */
+public class RtmpDataSource implements DataSource {
+
+
+    private final RtmpClient rtmpClient;
+    private Uri uri;
+
+    private RtmpDataSource() {
+        rtmpClient = new RtmpClient();
+    }
+
+    @Override
+    public Uri getUri() {
+        return uri;
```

Comment:
Hm. This is always null? I think it needs setting in `open` and clearing in `close`?